apps/typegpu-docs/src/content/docs/fundamentals/vertex-layouts.mdx

```diff
@@ -0,0 +1,196 @@
+---
+title: Vertex Layouts
+description: A guide on how to create and use typed vertex layouts
+---
+
+:::note[Recommended reading]
+We assume that you are familiar with the following concepts:
+- <a href="https://webgpufundamentals.org/webgpu/lessons/webgpu-vertex-buffers.html" target="_blank" rel="noopener noreferrer">Vertex Buffers</a>
+- <a href="https://www.w3.org/TR/webgpu/#vertex-state" target="_blank" rel="noopener noreferrer">WebGPU Vertex State</a>
+- <a href="https://www.w3.org/TR/WGSL/#input-output-locations" target="_blank" rel="noopener noreferrer">Input-output locations</a>
+- <a href="https://developer.mozilla.org/en-US/docs/Web/API/GPURenderPassEncoder" target="_blank" rel="noopener noreferrer">Render pass encoding</a>
+:::
+
+Vertex layouts are much like bind group layouts, in that they define the relationship between shaders and buffers. More precisely, they define what vertex attributes a shader expects, and how they are laid out in the corresponding vertex buffer.
+
+## Creating a vertex layout
+
+To create a vertex layout, use the `tgpu.vertexLayout` function. It takes an array schema constructor, i.e., a function that returns an array schema given the number of elements to render (vertices/instances). To determine what each element of the array corresponds to, you can pass an optional `stepMode` argument, which can be either `vertex` (default) or `instance`.
+
+```ts
+import tgpu from 'typegpu';
+import * as d from 'typegpu/data';
+
+const ParticleGeometry = d.struct({
+  tilt: d.f32,
+  angle: d.f32,
+  color: d.vec4f,
+});
+
+const geometryLayout = tgpu
+  .vertexLayout((n) => d.arrayOf(ParticleGeometry, n), 'instance');
+```
+
+## Utilizing loose schemas with vertex layouts
+
+The above example is great if the vertex buffer will also be used as a storage or uniform buffer. However, if you're only interested in the vertex data, you can use a loose schema instead.
+It is not restricted by alignment rules and can utilize many useful [vertex formats](https://www.w3.org/TR/webgpu/#vertex-formats). To create loose schemas, use `d.unstruct` instead of `d.struct` and `d.disarrayOf` instead of `d.arrayOf`.
+Inside of loose schemas you can use vertex formats as well as the usual data types.
```

Comment:
Thank you ChatGPU 🫡 
```suggestion
If the vertex buffer is not required to function as a storage or uniform buffer, a *loose schema* may be used to define the vertex data layout. Loose schemas are not subject to alignment restrictions and allow the use of various [vertex formats](https://www.w3.org/TR/webgpu/#vertex-formats).

To define a loose schema:
- Use `d.unstruct` instead of `d.struct`.
- Use `d.disarrayOf` instead of `d.arrayOf`.

Within a loose schema, both standard data types and vertex formats can be utilized.
```